### PR TITLE
ISPN-11576 doc fixes for dev guide content

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-11576

Needs backport to 10.1.x.

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/code_examples/AuthorMarshaller.java [hunk 1]
```diff
@@ -13,7 +13,7 @@ public String getTypeName() {
    }
 
    @Override
-   public void writeTo(ProtoStreamWriter writer, Author author) throws IOException {
+   public void writeTo(MessageMarshaller.ProtoStreamWriter writer, Author author) throws IOException {
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/code_examples/AuthorMarshaller.java:16]: @ryanemerson Can you please double check this change? There are a few other changes to the marshalling content.

File: documentation/src/main/asciidoc/topics/code_examples/BookMarshaller.java [hunk 2]
```diff
@@ -13,7 +13,7 @@ public String getTypeName() {
    }
 
    @Override
-   public void writeTo(ProtoStreamWriter writer, Book book) throws IOException {
+   public void writeTo(MessageMarshaller.ProtoStreamWriter writer, Book book) throws IOException {
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/code_examples/BookMarshaller.java:16]: @ryanemerson

File: documentation/src/main/asciidoc/topics/code_examples/CustomMarshaller.java [hunk 3]
```diff
@@ -1,3 +1,4 @@
 GlobalConfigurationBuilder builder = new GlobalConfigurationBuilder();
-      builder.serialization().marshaller(org.infinispan.example.marshall.CustomMarshaller.class)
-      .addJavaSerialWhiteList("org.infinispan.example.*");
+builder.serialization()
+      .marshaller(new org.infinispan.example.marshall.CustomMarshaller()).whiteList()
+      .addRegexp("org.infinispan.example.*");
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/code_examples/CustomMarshaller.java:4]: @ryanemerson

File: documentation/src/main/asciidoc/topics/code_examples/JBossUserMarshaller.java [hunk 4]
```diff
@@ -1,2 +1,2 @@
 GlobalConfigurationBuilder builder = new GlobalConfigurationBuilder();
-      builder.serialization().marshaller(JBossUserMarshaller.class);
+builder.serialization().marshaller(new JBossUserMarshaller());
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/code_examples/JBossUserMarshaller.java:2]: @ryanemerson

File: documentation/src/main/asciidoc/topics/code_examples/ProtoStreamMarshaller.java [hunk 5]
```diff
@@ -1,2 +1,3 @@
 GlobalConfigurationBuilder builder = new GlobalConfigurationBuilder();
-builder.addContextInitializers(new LibraryInitializerImpl(), new SCIImpl())
+builder.serialization()
+       .addContextInitializers(new LibraryInitializerImpl(), new SCIImpl());
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/code_examples/ProtoStreamMarshaller.java:3]: @ryanemerson

File: documentation/src/main/asciidoc/topics/code_examples/SerializationMarshaller.java [hunk 6]
```diff
@@ -1,3 +1,5 @@
 GlobalConfigurationBuilder builder = new GlobalConfigurationBuilder();
-builder.serialization().marshaller(new JavaSerializationMarshaller())
-      .addJavaSerialWhiteList("org.infinispan.example.*", "org.infinispan.concrete.SomeClass");
+builder.serialization()
+       .marshaller(new JavaSerializationMarshaller())
+       .whiteList()
+       .addRegexps("org.infinispan.example.", "org.infinispan.concrete.SomeClass");
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/code_examples/SerializationMarshaller.java:5]: @ryanemerson

File: documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc [hunk 7]
```diff
@@ -19,7 +19,7 @@ include::config_examples/custom_marshaller.xml[]
 [TIP]
 ====
 Custom marshaller implementations can access a configured white list via the
-link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html#initialize[initialize()] method, which is called during startup.
+link:{javadocroot}org/infinispan/commons/marshall/Marshaller.html#initialize(org.infinispan.commons.configuration.ClassWhiteList)[initialize()] method, which is called during startup.
 ====
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc:23]: @ryanemerson

File: documentation/src/main/asciidoc/topics/cache_api.adoc [hunk 8]
```diff
@@ -164,7 +164,7 @@ include::code_examples/ListenerMyNonBlocking.java[]
 ----
 
 ==== Asynchronous thread pool
-To tune the thread pool used to dispatch such asynchronous notifications, use the link:http://docs.jboss.org/infinispan/{infinispanversion}/configdocs/infinispan-config-{infinispanversion}.html[`<listener-executor />`] XML element in your configuration file.
+To tune the thread pool used to dispatch such asynchronous notifications, use the link:{configdocroot}[`<listener-executor />`] XML element in your configuration file.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/cache_api.adoc:167]: add the html file to the link?
```suggestion
To tune the thread pool used to dispatch such asynchronous notifications, use the link:{configdocroot}/infinispan-config-{infinispanversion}.html[`<listener-executor />`] XML element in your configuration file.
```

File: documentation/src/main/asciidoc/topics/cache_api.adoc [hunk 9]
```diff
@@ -192,26 +195,25 @@ These are, in order of cost:
 * writing to a cache store (optional)
 * locking
 
-As of {brandname} 4.0, using the async methods will take the network calls and marshalling off the critical path.  For various technical reasons, writing to a cache store and acquiring locks, however, still happens in the caller's thread.  In future, we plan to take these offline as well.  See link:http://lists.jboss.org/pipermail/infinispan-dev/2010-January/002219.html[this developer mail list thread] about this topic.
+Using the async methods will take the network calls and marshalling off the critical path.  For various technical reasons, writing to a cache store and acquiring locks, however, still happens in the caller's thread.
 
+//dnaro: ISPN-11577
+ifndef::productized[]
 === Notifying futures
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/cache_api.adoc:202]: this section can be removed IMO. We return CompletableFuture (a JDK class) and the users can use any of the methods available to register listeners.

File: documentation/src/main/asciidoc/topics/cache_api.adoc [hunk 10]
```diff
@@ -192,26 +195,25 @@ These are, in order of cost:
...
-
-==  Invocation Flags
-An important aspect of getting the most of {brandname} is the use of per-invocation flags in order to provide specific behaviour to each particular cache call. By doing this, some important optimizations can be implemented potentially saving precious time and network resources. One of the most popular usages of flags can be found right in Cache API, underneath the link:{javadocroot}/org/infinispan/Cache.html#putForExternalRead-K-V-[putForExternalRead()] method which is used to load an {brandname} cache with data read from an external resource. In order to make this call efficient, {brandname} basically calls a normal put operation passing the following flags: link:{javadocroot}/org/infinispan/context/Flag.html#FAIL_SILENTLY[FAIL_SILENTLY] , link:{javadocroot}/org/infinispan/context/Flag.html#FORCE_ASYNCHRONOUS[FORCE_ASYNCHRONOUS] , link:{javadocroot}/org/infinispan/context/Flag.html#ZERO_LOCK_ACQUISITION_TIMEOUT[ZERO_LOCK_ACQUISITION_TIMEOUT]
+== Invocation Flags
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/cache_api.adoc:210]: is this section supposed to be out of prod? 
you open `ifndef::productized[]` in the section before.

File: documentation/src/main/asciidoc/topics/cache_api.adoc (hunk 9 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/cache_api.adoc]: removed. thanks @pruivo

File: documentation/src/main/asciidoc/topics/dependencies_maven/jcache.xml [hunk 11]
```diff
@@ -3,3 +3,7 @@
   <artifactId>infinispan-jcache</artifactId>
   <version>${version.infinispan}</version>
 </dependency>
+<dependency>
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/dependencies_maven/jcache.xml:6]: I'm not sure about this... 
`cache-api` must be in the classpath if JCache is being used, independently of the provider used.

File: documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc [hunk 12]
```diff
@@ -19,7 +19,7 @@ include::config_examples/custom_marshaller.xml[]
 [TIP]
 ====
 Custom marshaller implementations can access a configured white list via the
-link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html#initialize[initialize()] method, which is called during startup.
+link:{javadocroot}org/infinispan/commons/marshall/Marshaller.html#initialize(org.infinispan.commons.configuration.ClassWhiteList)[initialize()] method, which is called during startup.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc:22]: change the link to point to github? that package is private and it isn't in javadocs.

File: documentation/src/main/asciidoc/topics/streams.adoc [hunk 13]
```diff
@@ -214,7 +212,7 @@ payload size even further.
 
 [source,java]
 ----
-   Map<Object, String> jbossValues = cache.entrySet().stream()
+Map<Object, String> map = (Map<Object, String>) cache.entrySet().stream()
               .filter(new ContainsFilter("Jboss"))
               .collect(ToMapCollectorSupplier.INSTANCE);
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/streams.adoc:217]: this collector doesn't exist.
I think you can use `.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));`

File: documentation/src/main/asciidoc/topics/streams.adoc [hunk 14]
```diff
@@ -292,7 +290,7 @@ replication timeout as a default timeout. You can set the timeout in your task b
 
 [source,java]
 ----
-CacheStream<Object, String> stream = cache.entrySet().stream();
+CacheStream<Object> stream = cache.entrySet().stream();
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/streams.adoc:293]:
```suggestion
CacheStream<Map.Entry<Object, String>> stream = cache.entrySet().stream();
```

File: documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc (hunk 12 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc:22]: let's remove this and the pointer to the Marshaller docs beneath it. I think we want to discourage custom marshallers if possible, right? maybe this should even just be community only.

File: documentation/src/main/asciidoc/topics/streams.adoc (hunk 13 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/streams.adoc:217]: fixed. thanks @pruivo

File: documentation/src/main/asciidoc/topics/dependencies_maven/jcache.xml (hunk 11 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/dependencies_maven/jcache.xml:6]: removed this from the pom example and added details about putting `cache-api` on the classpath.

File: documentation/src/main/asciidoc/topics/code_examples/MultiMapCache.java [hunk 15]
```diff
@@ -1,4 +1,4 @@
-public interface MultimapCache<K, V> {
+public interface BasicMultimapCache<K, V> {
```
[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/code_examples/MultiMapCache.java:1]: @pruivo `CompletableFuture<Void> remove(Predicate<? super V> p);` is already there but I've added the other two and reverted the name change.

File: documentation/src/main/asciidoc/topics/code_examples/EncoderScrambler.java [hunk 16]
```diff
@@ -1,15 +1,24 @@
-public class Scrambler implements Encoder {
+class Scrambler implements Encoder {
 
-   Object toStorage(Object content) {
-      // Encrypt data
-   }
+        public Object toStorage(Object content) {
```
[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/code_examples/EncoderScrambler.java:3]: fixed. thanks @pruivo

File: documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc (hunk 12 above)
Comment (pruivo) [reply documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc]: for Marshaller question, I'll invoke @ryanemerson  :)

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 17]
```diff
@@ -301,14 +292,14 @@ for the metadata parameters already provided by the functional map API.
...
+link:{javadocroot}/org/infinispan/functional/Param.html[Per-invocation parameters]
 are applied to regular functional map API calls to
 alter the behaviour of certain aspects. Adding per invocation parameters is
 done using the
-link:{javadocroot}/org/infinispan/commons/api/functional/FunctionalMap.html#withParams-org.infinispan.commons.api.functional.Param...-[`withParams(Param<?>...)`]
+link:{javadocroot}/org/infinispan/functional/FunctionalMap.html#withParams-org.infinispan.commons.api.functional.Param...-[`withParams(Param<?>...)`]
 method.
 
-link:{javadocroot}/org/infinispan/commons/api/functional/Param.FutureMode.html[`Param.FutureMode`]
+link:{javadocroot}/org/infinispan/functional/Param.FutureMode.html[`Param.FutureMode`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:302]: FutureMode has been removed. Paragraph and example can be removed.

File: documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc (hunk 12 above)
Comment (ryanemerson) [reply documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc]: @pruivo The package is public https://github.com/infinispan/infinispan/blob/4139c97c1a34c65bbd28b1e24467a543af9aa04f/commons/all/src/main/java/org/infinispan/commons/marshall/package-info.java#L7-L6

I don't think we should dissuade the use of custom marshallers. On the client they can be very useful and have been used in the past by users.

File: documentation/src/main/asciidoc/topics/functional_api.adoc (hunk 17 above)
Comment (pruivo) [reply documentation/src/main/asciidoc/topics/functional_api.adoc:302]: The next paragraph (about `Param.PersistenceMode`) needs to be revisited (package and links need to be updated)

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 18]
```diff
@@ -378,22 +369,22 @@ entry has been written.
 However, write event listeners can distinguish between entry removals
 and cache entry create/modify-update events because they can query
 what the new entry's value via
-link:{javadocroot}/org/infinispan/commons/api/functional/EntryView.ReadEntryView.html#find--[`ReadEntryView.find()`]
+link:{javadocroot}/org/infinispan/functional/EntryView.ReadEntryView.html#find--[`ReadEntryView.find()`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:372]:
```suggestion
link:{javadocroot}/org/infinispan/functional/EntryView.ReadEntryView.html#find()[`ReadEntryView.find()`]
```

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 19]
```diff
@@ -378,22 +369,22 @@ entry has been written.
...
 method.
 
 Adding a write listener is done via the WriteListeners interface
 which is accessible via both
-link:{javadocroot}/org/infinispan/commons/api/functional/FunctionalMap.ReadWriteMap.html#listeners--[`ReadWriteMap.listeners()`]
+link:{javadocroot}/org/infinispan/functional/FunctionalMap.ReadWriteMap.html#listeners--[`ReadWriteMap.listeners()`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:377]:
```suggestion
link:{javadocroot}/org/infinispan/functional/FunctionalMap.ReadWriteMap.html#listeners()[`ReadWriteMap.listeners()`]
```

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 20]
```diff
@@ -378,22 +369,22 @@ entry has been written.
...
 and
-link:{javadocroot}/org/infinispan/commons/api/functional/FunctionalMap.WriteOnlyMap.html#listeners--[`WriteOnlyMap.listeners()`]
+link:{javadocroot}/org/infinispan/functional/FunctionalMap.WriteOnlyMap.html#listeners--[`WriteOnlyMap.listeners()`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:379]:
```suggestion
link:{javadocroot}/org/infinispan/functional/FunctionalMap.WriteOnlyMap.html#listeners()[`WriteOnlyMap.listeners()`]
```

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 21]
```diff
@@ -378,22 +369,22 @@ entry has been written.
...
  method.
 
 A write listener implementation can be defined either passing a function
 to
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.WriteListeners.html#onWrite-java.util.function.Consumer-[`onWrite(Consumer<ReadEntryView<K, V>>)`]
+link:{javadocroot}/org/infinispan/functional/Listeners.WriteListeners.html#onWrite-java.util.function.Consumer-[`onWrite(Consumer<ReadEntryView<K, V>>)`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:384]:
```suggestion
link:{javadocroot}/org/infinispan/functional/Listeners.WriteListeners.html#onWrite(java.util.function.Consumer)[`onWrite(Consumer<ReadEntryView<K, V>>)`]
```

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 22]
```diff
@@ -378,22 +369,22 @@ entry has been written.
...
 method, or passing a
 WriteListener implementation to
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.WriteListeners.html#add-org.infinispan.commons.api.functional.Listeners.WriteListeners.WriteListener-[`add(WriteListener<K, V>)`]
+link:{javadocroot}/org/infinispan/functional/Listeners.WriteListeners.html#add-org.infinispan.commons.api.functional.Listeners.WriteListeners.WriteListener-[`add(WriteListener<K, V>)`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:387]:
```suggestion
link:{javadocroot}/org/infinispan/functional/Listeners.WriteListeners.html#add(org.infinispan.functional.Listeners.WriteListeners.WriteListener)[`add(WriteListener<K, V>)`]
```

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 23]
```diff
@@ -417,18 +408,18 @@ differentiation between create, modified and removed can be fully
 guaranteed.
 
 Adding a read-write listener is done via the
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.ReadWriteListeners.html[`ReadWriteListeners`]
+link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.html[`ReadWriteListeners`]
 interface which is accessible via
-link:{javadocroot}/org/infinispan/commons/api/functional/FunctionalMap.ReadWriteMap.html#listeners--[`ReadWriteMap.listeners()`]
+link:{javadocroot}/org/infinispan/functional/FunctionalMap.ReadWriteMap.html#listeners--[`ReadWriteMap.listeners()`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:413]:
```suggestion
link:{javadocroot}/org/infinispan/functional/FunctionalMap.ReadWriteMap.html#listeners()[`ReadWriteMap.listeners()`]
```

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 24]
```diff
@@ -417,18 +408,18 @@ differentiation between create, modified and removed can be fully
...
 method.
 
 If interested in only one of the event types, the simplest way to add a
 listener is to pass a function to either
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onCreate-org.infinispan.commons.api.functional.EntryView.ReadEntryView-[`onCreate`]
+link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onCreate-org.infinispan.commons.api.functional.EntryView.ReadEntryView-[`onCreate`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:418]:
```suggestion
link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onCreate(java.util.function.Consumer)[`onCreate`]
```

File: documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc (hunk 12 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc]: @pruivo I've reverted the changes and verified that it's available in javadocs. https://access.redhat.com/webassets/avalon/d/red-hat-data-grid/8.0/api/org/infinispan/commons/marshall/Marshaller.html#initialize(org.infinispan.commons.configuration.ClassWhiteList)

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 25]
```diff
@@ -417,18 +408,18 @@ differentiation between create, modified and removed can be fully
...
 ,
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onModify-org.infinispan.commons.api.functional.EntryView.ReadEntryView-org.infinispan.commons.api.functional.EntryView.ReadEntryView-[`onModify`]
+link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onModify-org.infinispan.commons.api.functional.EntryView.ReadEntryView-org.infinispan.commons.api.functional.EntryView.ReadEntryView-[`onModify`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:420]:
```suggestion
link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onModify(java.util.function.BiConsumer)[`onModify`]
```

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 26]
```diff
@@ -417,18 +408,18 @@ differentiation between create, modified and removed can be fully
...
 or
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onRemove-org.infinispan.commons.api.functional.EntryView.ReadEntryView-[`onRemove`]
+link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onRemove-org.infinispan.commons.api.functional.EntryView.ReadEntryView-[`onRemove`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:422]:
```suggestion
link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.ReadWriteListener.html#onRemove(java.util.function.Consumer)[`onRemove`]
```

Comment (pruivo) [reply documentation/src/main/asciidoc/topics/functional_api.adoc:422]: ps. examples need to be updated

File: documentation/src/main/asciidoc/topics/functional_api.adoc [hunk 27]
```diff
@@ -439,9 +430,9 @@ include::code_examples/ReadWriteMapAutoCloseable.java[]
 
 If listening for two or more event types, it's better to pass in an
 implementation of
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.WriteListeners.WriteListener.html[`ReadWriteListener`]
+link:{javadocroot}/org/infinispan/functional/Listeners.WriteListeners.WriteListener.html[`ReadWriteListener`]
 interface via the
-link:{javadocroot}/org/infinispan/commons/api/functional/Listeners.ReadWriteListeners.html#add-org.infinispan.commons.api.functional.Listeners.ReadWriteListeners.ReadWriteListener-[`ReadWriteListeners.add()`]
+link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.html#add-org.infinispan.commons.api.functional.Listeners.ReadWriteListeners.ReadWriteListener-[`ReadWriteListeners.add()`]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/functional_api.adoc:435]:
```suggestion
link:{javadocroot}/org/infinispan/functional/Listeners.ReadWriteListeners.html#add(org.infinispan.functional.Listeners.ReadWriteListeners.ReadWriteListener)[`ReadWriteListeners.add()`]
```

File: documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc (hunk 12 above)
Comment (pruivo) [reply documentation/src/main/asciidoc/topics/proc_using_custom_marshallers.adoc]: no idea what link I used xD eh eh

File: documentation/src/main/asciidoc/topics/cache_api.adoc [hunk 28]
```diff
@@ -164,7 +164,7 @@ include::code_examples/ListenerMyNonBlocking.java[]
 ----
 
 ==== Asynchronous thread pool
-To tune the thread pool used to dispatch such asynchronous notifications, use the link:http://docs.jboss.org/infinispan/{infinispanversion}/configdocs/infinispan-config-{infinispanversion}.html[`<listener-executor />`] XML element in your configuration file.
+To tune the thread pool used to dispatch such asynchronous notifications, use the link:{configdocroot}/infinispan-config-{infinispanversion}.html[`<listener-executor />`] XML element in your configuration file.
 
 [[cache_asynchronous_api]]
 ==  Asynchronous API
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/cache_api.adoc:170]: links need to be updated in this subsection. 
Also, can you double-check the text? We no longer return a Future but a CompletableFuture.

**Comment by pruivo** [comment]
integrated! thanks @oraNod !

**Pull request merged**